pytorch_lightning/metrics/functional/classification.py

```diff
@@ -397,3 +398,34 @@ def dice_score(pred: torch.Tensor, target: torch.Tensor, bg: bool = False,
 
         scores[i] += score_cls
     return reduce(scores, reduction=reduction)
+
+
+def iou(pred: torch.Tensor, target: torch.Tensor, remove_bg: bool = False,
+        reduction: str = 'elementwise_mean'):
+    """
+    Intersection over union, or Jaccard index calculation.
+
+    Args:
+        pred: Tensor containing predictions
+
+        target: Tensor containing targets
+
+        remove_bg: Flag to state whether a background class has been included
+            within input parameters. If true, will remove background class. If
+            false, return IoU over all classes.
+            Assumes that background is '0' class in input tensor
+
+        reduction: Reduction to be performed on output
+
+    Returns:
+        IoU score : Tensor containing single value if reduction is
+            'elementwise_mean', or number of classes if reduction is 'none'
+    """
+    cm = confusion_matrix(pred, target, normalize=True)
+    if remove_bg:
+        cm = cm[1:, 1:]
+    tp = torch.diag(cm)
+    fp = cm.sum(dim=0) - tp
```

Comment:
we basically also have a function to calculate these values without calculating a whole cm

Reply:
Updated - Missed these first time, sorry!